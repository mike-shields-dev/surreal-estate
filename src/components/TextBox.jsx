import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/TextBox.module.css";

const TextBox = ({ field, maxLength, onChange, required, type, value }) => {
  return (
    <div className={styles.textbox}>
      <label htmlFor={field} className={styles.textbox__label}>
        {field}
      </label>
      <input
        required={required}
        id={field}
        name={field}
        onChange={onChange}
        type={type}
        value={value}
        maxLength={maxLength}
      />
    </div>
  );
};

TextBox.defaultProps = {
  maxLength: null,
  required: false,
};

TextBox.propTypes = {
  field: PropTypes.string.isRequired,
  maxLength: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

export default TextBox;
