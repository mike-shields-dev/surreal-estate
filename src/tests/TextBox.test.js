import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import TextBox from "../components/TextBox";

const props = {
  onChange: jest.fn(),
  field: "title",
  type: "text",
  value: "",
};

const renderTextBox = () => render(<TextBox {...props} />);

describe("TextBox", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderTextBox();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders text input with associated label and given props", () => {
    renderTextBox();

    const textInput = screen.getByLabelText(props.field);

    expect(textInput.getAttribute("id")).toEqual(props.field);
    expect(textInput.getAttribute("name")).toEqual(props.field);
    expect(textInput.getAttribute("type")).toEqual(props.type);
    expect(textInput).toHaveValue(props.value);
  });

  it("invokes the onChange callback", () => {
    renderTextBox();
    const textInput = screen.getByRole("textbox", { name: props.field });

    fireEvent.change(textInput, {
      target: { value: "Test" },
    });

    expect(props.onChange).toHaveBeenCalled();
  });
});
