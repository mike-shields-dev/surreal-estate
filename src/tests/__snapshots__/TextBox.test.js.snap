// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="textbox"
  >
    <label
      class="textbox__label"
      for="title"
    >
      title
    </label>
    <input
      id="title"
      maxlength="64"
      name="title"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
